typed/measures-with-dimensions/chemistry/element-struct.rkt

```diff
@@ -19,10 +19,10 @@ require racket/match
   ([atomic-number : Positive-Integer] [symbol : Sym] [atomic-mass : Molar-Mass])
   #:transparent
   #:property prop:custom-write
-  (lambda ([e : (element Any)] [out : Output-Port] [mode : (Un 0 1 #t #f)])
-    (match mode
-      [(or 0 #f) (fprintf out "~a" (element-symbol e))]
-      [(or 1 #t) (fprintf out "#<element:~a>" (element-symbol e))]))
+  (lambda ([e : (element Any)] [out : Output-Port] [mode : (Un Boolean 0 1)])
+    (case mode
+      [(1 #t) (fprintf out "#<element:~a>" (element-symbol e))]
+      [(0 #f) (fprintf out "~a" (element-symbol e))]))
```

Comment:
Why change this? The type `(Un 0 1 #t #f)` is equivalent, and the body seems to type-check fine as it is.